cel/env.go

```diff
@@ -405,6 +405,12 @@ func (e *Env) ResidualAst(a *Ast, details *EvalDetails) (*Ast, error) {
 	return checked, nil
 }
 
+// EstimateCost estimates the cost of the CEL expression using the length estimates of input data and extension functions
+// provided by estimator.
+func (e *Env) EstimateCost(ast *exprpb.CheckedExpr, estimator checker.CostEstimator) checker.CostEstimate {
```

Comment:
@TristonianJones Should we include something like this? Would you expect it to automatically make sure the checker is run?

Reply:
Nevermind, I think I figured out how to do this reasonably cleanly.